packages/dropdowns/src/Select/Select.tsx

```diff
@@ -58,6 +68,85 @@ export const Select = React.forwardRef<HTMLDivElement, ISelectProps>(
       previousIsOpenRef.current = isOpen;
     }, [isOpen, triggerRef]);
 
+    /**
+     * Handle timeouts for clearing search text
+     */
+    useEffect(() => {
+      // Cancel existing timeout if keys are pressed rapidly
+      if (searchTimeoutRef.current) {
+        clearTimeout(searchTimeoutRef.current);
+      }
+
+      // Reset search string after delay
+      searchTimeoutRef.current = (setTimeout(() => {
+        setSearchString('');
+      }, 500) as unknown) as number;
```

Comment:
An alternative to casting the timeout's return value as `unknown` and then to `number` would be to call `setTimeout` off of the window property. I think this tells TypeScript its a browser environment timeout. It then uses the correct browser timeout type definitions, rather than the NodeJS timeout type definitions.

```tsx
searchTimeoutRef.current = window.setTimeout(() => {
  setSearchString('');
}, 500);
```

Reply:
TIL that's really nice. Updated.